app/src/main/java/de/danoeh/antennapod/fragment/swipeactions/SwipeActions.java

```diff
@@ -0,0 +1,282 @@
+package de.danoeh.antennapod.fragment.swipeactions;
+
+import android.content.Context;
+import android.content.SharedPreferences;
+import android.graphics.Canvas;
+
+import androidx.annotation.NonNull;
+import androidx.core.content.ContextCompat;
+import androidx.core.graphics.ColorUtils;
+import androidx.fragment.app.Fragment;
+import androidx.lifecycle.Lifecycle;
+import androidx.lifecycle.LifecycleEventObserver;
+import androidx.recyclerview.widget.ItemTouchHelper;
+import androidx.recyclerview.widget.RecyclerView;
+
+import com.annimon.stream.Stream;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.List;
+
+import de.danoeh.antennapod.R;
+import de.danoeh.antennapod.dialog.SwipeActionsDialog;
+import de.danoeh.antennapod.fragment.EpisodesFragment;
+import de.danoeh.antennapod.fragment.QueueFragment;
+import de.danoeh.antennapod.model.feed.FeedItem;
+import de.danoeh.antennapod.model.feed.FeedItemFilter;
+import de.danoeh.antennapod.ui.common.ThemeUtils;
+import de.danoeh.antennapod.view.viewholder.EpisodeItemViewHolder;
+import it.xabaras.android.recyclerview.swipedecorator.RecyclerViewSwipeDecorator;
+
+public class SwipeActions {
+    public static final String PREF_NAME = "SwipeActionsPrefs";
+    public static final String PREF_SWIPEACTIONS = "PrefSwipeActions";
+    public static final String PREF_NO_ACTION = "PrefsNoSwipeAction";
+
+    public static final List<SwipeAction> swipeActions = Collections.unmodifiableList(
+            Arrays.asList(new AddToQueueSwipeAction(), new MarkUnplayedSwipeAction(),
+                    new StartDownloadSwipeAction(), new MarkFavouriteSwipeAction(),
+                    new MarkPlayedSwipeAction(), new RemoveFromQueueSwipeAction())
+    );
+
+    private RecyclerView recyclerView;
+    public ItemTouchHelper itemTouchHelper;
+    private final Fragment fragment;
+    private final String tag;
+    private FeedItemFilter filter = null;
+
+    private NewSwipeCallback newSwipeCallback = SimpleSwipeCallback::new;
```

Comment:
I don't understand why this is called "new". Was first thinking about the "new" screen but it seems unrelated.

Reply:
this is creating a new one, this is requiered for queue

Reply:
Can't you update/notify the existing one instead of creating a new one? Supplying a `factory` for swipe actions seems to be an unnecessary layer of indirection to me.

Reply:
I don't think so, but feel free to try, I think it said variables in simplecallback need to be effectively final, so they can't be updated from outside.